li-lucia/TerpsiKo
# Aws postgres connection

# Description
This pull request connects the express server to TerpsiKo's AWS RDS Postgres database instance. It establishes the necessary database connection settings and integrates them into the existing server setup. A test route has been added to test that the connection works. All firebase and firbase/auth related configuration has been removed from the server and from the client folders.

## Type of change


## Changes Made
- Added **.env** and AWS' SSL certificate **us-east-1-bundle.cert** to **.gitignore**
- Deleted **.gitignore** within the server file as the one in the root directory is sufficient
- Removed firebase auth functionality from the **LoginDisplay** component
- Uninstalled firebase packages from client and server package.json
- Installed **pg** package to interact with postgres database
- Created **db.js** file that creates a connection pool to the database instance and exports a function **runquery** that acquires a connection to the pool and executes a query
- Implemented a test route and test controller that fetches all entries from the **users** table

# Testing Instructions

1. Navigate to the **server** directory
2. Install all packages by running **npm install**
3. Place AWS' SSL certificate **us-east-1-bundle.cert** into this directory
4. Create a **.env** file and set the necessary environment variables including **PORT**, **DB_USERNAME**, **DB_RDS_ENDPOINT**, **DB_NAME**, **DB_PASSWORD**, and **DB_PORT**.
6. Start the server using **npm run dev**
7. Access the test route **/test** using Postman or URL and ensure that you are getting all entries from the **users** table

# Checklist:

- [x] I have followed the coding style and guidelines of this project.
- [x] I have reviewed my own code to ensure its quality.
- [x] I have added relevant comments and documentation where necessary.
- [ ] I have updated/added necessary tests for the changes made.
- [x] I have rebased my branch on top of the latest main (or other target) branch.
